engine/src/test/java/com/redhat/cloud/notifications/processors/email/EmailAggregatorTest.java

```diff
@@ -63,13 +66,21 @@ class EmailAggregatorTest {
     @InjectMock
     EndpointRepository endpointRepository;
 
+    final EmailAggregationKey aggregationKey = new EmailAggregationKey("org-1", "rhel", "policies");
+
     @AfterEach
     void afterEach() {
         featureFlipper.setUseEventTypeForSubscriptionEnabled(false);
     }
 
+    @BeforeEach
+    void beforeEach() {
+        emailAggregator.aggregationMaxPageSize = 5;
```

Comment:
It doesn't look like the test in this class triggers multiple iterations of the new `do while` loop that loads the aggregation from the DB. Did I read it wrong?

Reply:
Well it does but it's not obvious. I added a comment and some Mockito verify methods to clarify it.